flamedisx/xenon/x1t_sr1.py

```diff
@@ -135,13 +135,11 @@ def cal_rec_efficiency_tf(sig, fmap, domain_def, pivot_pt):
     sig_tf = tf.convert_to_tensor(sig, dtype=fd.float_type())
     bias_median = interpolate_tf(sig_tf, fmap[1], domain_def)
 
-    if pivot_pt<0:
-        bias_other = interpolate_tf(sig_tf, fmap[0], domain_def)
-        bias_out = pivot_pt*(bias_median-bias_other)+bias_median
-    else:
-        bias_other = interpolate_tf(sig_tf, fmap[2], domain_def)
-        bias_out = pivot_pt*(bias_other-bias_median)+bias_median
-    return bias_out
+    bias_diff = tf.cond(
```

Comment:
I considered using `tf.cond` while writing this (along with using `tf.math.multiply` instead of `*` and so on) just for explicitness. However, I decided against it in the end because the rest of flamedisx does not stick to strict explicitness when it comes to `tf` functions.

In this case, the 'if' statement is also used in other fundamental blocks that are traced over. (Just one example, but did not verify if there are more: https://github.com/FlamTeam/flamedisx/blob/e17ecdbfc841e53a82b21c7672698c16a49ba75b/flamedisx/lxe_blocks/quanta_splitting.py#L51) Replacing `if` with `tf.cond` might circumvent the issue at this point, but will be caught downstream in other traced functions in other blocks.

Reply:
The problem comes from the `if` statement being used on a tensor, whose value can change during the fit. There are of course many if statements in flamedisx, but as far as I can see, we always use `tf.where`, `tf.clip` etc. instead of `if` when dealing with tensors. There are several examples of this in `x1t_sr1.py`.

Reply:
Just for the sake of documentation so that I understand this issue in the future.

The issue is with using `if` on `pivot_pt` which is a tensor and its value can change during the process of fitting. The different between this and https://github.com/FlamTeam/flamedisx/blob/e17ecdbfc841e53a82b21c7672698c16a49ba75b/flamedisx/lxe_blocks/quanta_splitting.py#L51
is that `self.do_pel_fluct` is just a 'flag' and does not change during the fitting process.